pkg/kubectl/cmd/create_podsecuritypolicy.go

```diff
@@ -0,0 +1,442 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/api/core/v1"
+	pspv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgo "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	PSPLong = templates.LongDesc(i18n.T(`
+		Create a PodSecurityPolicy.`))
+
+	pspExample = templates.Examples(i18n.T(`
+
+		# Create a PodSecurityPolicy named "example" that simply prevents the creation of privileged pods.
+		kubectl create podsecuritypolicy example --privileged=false --selinux=RunAsAny --supplemental-groups=RunAsAny
+		--run-as-user=RunAsAny --fs-group=RunAsAny
+
+		# Create a PodSecurityPolicy named "privileged"
+		kubectl create podsecuritypolicy privileged --privileged=true --allowed-cap=* --host-ports=0-65535
+		--selinux=RunAsAny --supplemental-groups=RunAsAny --run-as-user=RunAsAny --fs-group=RunAsAny
+
+		# Create a PodSecurityPolicy named "my-psp"
+		kubectl create podsecuritypolicy my-psp --selinux="MustRunAs,user=u,role=r,type=t,level=l" --supplemental-groups=RunAsAny
+		--run-as-user=RunAsAny --fs-group=RunAsAny
+
+		# Create a PodSecurityPolicy named "run-as-user-example" that allow user ID 10-100 containers run as.
+		kubectl create podsecuritypolicy run-as-user-example --selinux=RunAsAny --supplemental-groups=RunAsAny
+		--fs-group=RunAsAny --run-as-user="MustRunAs,10-100"
+		`))
+)
+
+type CreatePSPOptions struct {
```

Comment:
I think this exposes too many options... the goal of create commands is not to connect a CLI flag to every field in the object, but to provide a minimal/reasonable set of options as a starting point for a valid object.  See https://github.com/kubernetes/community/blob/master/contributors/devel/kubectl-conventions.md#create-commands

Reply:
@liggitt , thanks for your reply, I will fix this problem.
Shall we only remain SELinux,RunAsUser,SupplementalGroups ,FSGroup?
Since they are required options.

Reply:
> Shall we only remain SELinux,RunAsUser,SupplementalGroups ,FSGroup? Since they are required options.

Note that there is a plan to turn them into optional: https://github.com/kubernetes/kubernetes/issues/56174

CC @tallclair